src/heratape/files.py

```diff
@@ -303,3 +304,26 @@ def update_file(
     stmt = update(Files).where(Files.filebase == filebase).values(**update_vals)
     with HTSessionWrapper(session=session, testing=testing) as ht_sess:
         ht_sess.execute(stmt)
+def list_incomplete(tape_id=None, testing=False):
+    # list the names of files with no write date
+    # set tape_id =None to list all, or to a tape serial number to list those
+    # return a list of files that weren't signed off with a write date.
+    #  this can happen if the write was interrupted before the db could be informed of success
+    #  writes can take hours, so this unhappy situation is likely
+    with HTSessionWrapper(testing=testing) as ht_sess: 
+        if tape_id is None:
+            return ht_sess.query(func.distinct(Files.jd)).where(Files.write_date == None).all(), \
```

Comment:
This can be written instead as:

```suggestion
            return ht_sess.query(Files.jd).distinct().where(Files.write_date == None).all(), \
```

which I find slightly easier to read.

Reply:
This also applies 2 lines down